devtools/datasette/publish.py

```diff
@@ -110,7 +110,12 @@ def metadata(pudl_out: Path) -> str:
     flag_value="metadata",
     help="Generate the Datasette metadata.yml in current directory, but do not deploy.",
 )
-def deploy_datasette(deploy: str) -> int:
+@click.option(
```

Comment:
I thought about some sort of `publish.py -- <whatever options you want to pass all the way through to flyctl>` pattern but 2 minutes of reading the click docs didn't show me how to do it, so I figured it wasn't worth the time. If you know how to do that off the top of your head I'd gladly use that instead.

Reply:
I think anything after `--` it'll stop parsing options and [assume that all subsequent strings are arguments](https://click.palletsprojects.com/en/8.1.x/arguments/#option-like-arguments), so I think to capture those arguments and pass them in you need to define an [optional variadic argument](https://click.palletsprojects.com/en/8.1.x/arguments/#variadic-arguments) that'll consume all subsequent arguments and provide them as a tuple of strings, which would look something like...

```py
@click.argument(
    "fly_args",
    required=False,
    nargs=-1,
    help="Additional options or arguments which are passed through to flyctl deploy",
)
def deploy_datasette(deploy: str, fly_args: tuple) -> int:
    ...
    cmd = ["/usr/bin/env", "flyctl", "deploy"] + list(fly_args)
```